lib/miq_automation_engine/engine/miq_ae_builtin_method.rb

```diff
@@ -47,11 +57,12 @@ def self.powershell(_obj, inputs)
 
     def self.miq_parse_provider_category(obj, _inputs)
       provider_category = nil
-      keys = %w(vm orchestration_stack miq_request miq_provision miq_host_provision vm_migrate_task platform_category)
-      keys.detect { |k| provider_category = category_for_key(obj, k) }
+      ATTRIBUTE_LIST.detect { |attr| provider_category = category_for_key(obj, attr) }
```

Comment:
The `detect` method returns on the first result that is not false.  You want to set `provider_category` outside of the block.
`provider_category = ATTRIBUTE_LIST.detect { |attr| category_for_key(obj, attr) }`